src/Elmah.Io.AspNetCore/ElmahIoMiddleware.cs

```diff
@@ -8,44 +8,54 @@ namespace Elmah.Io.AspNetCore
     public class ElmahIoMiddleware
     {
         private readonly RequestDelegate _next;
+        private readonly string _apiKey;
+        private readonly Guid _logId;
         private readonly ElmahIoSettings _settings;
 
-        public static string ApiKey { get; private set; }
-        public static Guid LogId { get; private set; }
-
+        [Obsolete]
         public ElmahIoMiddleware(RequestDelegate next, string apiKey, Guid logId) : this(next, apiKey, logId, new ElmahIoSettings())
         {
         }
 
+        [Obsolete]
         public ElmahIoMiddleware(RequestDelegate next, string apiKey, Guid logId, ElmahIoSettings settings)
         {
             _next = next;
-            ApiKey = apiKey.AssertApiKey();
-            LogId = logId.AssertLogId();
+            _apiKey = apiKey.AssertApiKey();
+            _logId = logId.AssertLogId();
             _settings = settings.AssertSettings();
         }
 
+        public ElmahIoMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
         public async Task Invoke(HttpContext context)
         {
+            var config = (ElmahIoConfiguration)context.RequestServices.GetService(typeof(ElmahIoConfiguration));
+            var apiKey = config != null ? config.ApiKey : _apiKey;
+            var logId = config != null ? config.LogId : _logId;
+            var settings = config != null ? config.Settings : _settings;
```

Comment:
Make sure that every constructors provides a configuration object properly to avoid these checks.